server/base/src/main/java/org/apache/accumulo/server/fs/FileManager.java

```diff
@@ -386,6 +388,7 @@ private void releaseReaders(KeyExtent tablet, List<FileSKVIterator> readers,
 
   }
 
+  @NotThreadSafe
   static class FileDataSource implements DataSource {
```

Comment:
A supporting class for SourceSwitchingIterator which is an iterator which is not thread safe, and impl shows this is intended for single threaded access (e.g., no volatile, sync, etc).